File: C++/172.cpp
__________________________________________________________________________________________________
sample 4 ms submission
class Solution {
public:
    int trailingZeroes(int n) {
     
        
        int count=0;
        long long i=5;
        while(n/i)
        {
            count+=n/i;
            i=i*5;
        }
        
        return count;
        
    }
};
__________________________________________________________________________________________________
sample 8232 kb submission
class Solution {
public:
    int trailingZeroes(int n) {
        if(n<5) return 0;
        ios::sync_with_stdio(false);
        cin.tie(0);
        cout.tie(0);
        int ans = 0;
        for(long i=5 ; i<=n ; i+=5){
            if(i>n) break;
            int temp = i;
            while(temp%5 == 0){
                ans++;
                temp/=5;
            }
        }
        return ans;
    }
};
__________________________________________________________________________________________________

File: Java/172.java
__________________________________________________________________________________________________
sample 0 ms submission
class Solution {
    // 非递归解法
    public static int trailingZeroes(int n) {
        if (n<5)return 0;

        if (n<10)return 1;

        return (n/5+ trailingZeroes(n/5));
    }

}
__________________________________________________________________________________________________
sample 31896 kb submission
class Solution {
    public int trailingZeroes(int n) {
          int sum=5;        int count = 0;
        for (int i = 0; i < n; i++) {
               if(n/sum>0){
                   count +=n/sum;
                   sum *=5;
                   if(sum >1220703125){
                      break;
                  }
               }else{
                   break;
               }
        }
        return count;
    }
}
__________________________________________________________________________________________________
